ecosystem/platform/server/app/controllers/settings_controller.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+class SettingsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_user
+
+  layout 'settings'
+
+  def profile
+    return unless request.method_symbol == :post
+
+    user_params = params.fetch(:user, {}).permit(:username, :email)
+    if @user.update(user_params)
+      notice = if @user.pending_reconfirmation?
+                 "A verification email has been sent to #{@user.unconfirmed_email}."
+               else
+                 'User profile updated'
+               end
+      redirect_to settings_profile_url, notice:
+    else
+      render :profile, status: :unprocessable_entity
+    end
+  end
+
+  def connections
+    store_location_for(current_user, request.path)
+    @authorizations = @user.authorizations.to_h do |authorization|
+      [authorization.provider, authorization]
+    end
+
+    return unless request.method_symbol == :post
```

Comment:
this is enforceable via rails routes, right?

Reply:
Yeah it is probably a better idea to have separate methods for showing & updating the settings. Updated.